src/Network/WebSockets/Connection.hs

```diff
@@ -212,23 +225,32 @@ data ConnectionOptions = ConnectionOptions
     { connectionOnPong :: !(IO ())
       -- ^ Whenever a 'pong' is received, this IO action is executed. It can be
       -- used to tickle connections or fire missiles.
+    , connectionPermessageDeflate :: Maybe PermessageDeflate
+      -- ^ Settings of permessage deflate extension.
+      -- Nothing will disable it
     }
 
 
 --------------------------------------------------------------------------------
 defaultConnectionOptions :: ConnectionOptions
 defaultConnectionOptions = ConnectionOptions
     { connectionOnPong = return ()
+    , connectionPermessageDeflate = Just defaultPermessageDeflate
```

Comment:
That is a bit suspicious.  What happens if you use a `NOINLINE` pragma on `defaultConnectionOptions` and `defaultPermessageDeflate`?  That should allow us to share these in between connections.